libbeat/outputs/logstash/sync.go

```diff
@@ -114,6 +114,7 @@ func (c *syncClient) Publish(_ context.Context, batch publisher.Batch) error {
 	}
 
 	for len(events) > 0 {
+		begin := time.Now()
```

Comment:
I think this is too large of a span, we have reconnect and windowed events in here.  I think we need to re-think this part.  We really just want the time span to start right before the data is sent over the network, and end right when we know if data is written (ACK), even if there is an error we want to "bill" the time spent communicating with logstash.

FYI it's OK if we want to pull out logstash from this PR and make a new issue to track that going forward.

Reply:
Yah. I actually kinda chose that span deliberately, having reconnect be part of the count actually seemed fairly accurate? It's a blocking operation, so a reconnect would would influence the latency?

Reply:
It definitely increases the overall latency for sending the event, but that isn't what this metric is focused on.  This is just trying to get a sense of how long does it take for the output to receive the data and then get back to us with acknowledgment that the data is written.

An example might help, if the events/sec drop, but this metric goes up, we have a good indicator that we should be looking at the output (logstash/elasticsearch/kafka) for issues, not the beat.

Reply:
Alright, that makes sense. Tinkered with it a bit.